Dockerfile

```diff
@@ -1,13 +1,33 @@
-FROM ubuntu:16.04
+# Multi-stage docker build
+# Build stage
+FROM golang:1.14 AS builder
 
-ARG TARGETARCH
-ENV EXPORTER=/exporter-${TARGETARCH}
+LABEL maintainer="LitmusChaos"
 
-RUN apt-get update && apt-get install -y ca-certificates && rm -rf /var/cache/apk/*
+# LD_FLAGS is passed as argument from Makefile. It will be empty, if no argument passed
+ARG TARGETPLATFORM
 
-COPY .$EXPORTER /
+ADD . /chaos-exporter
+WORKDIR /chaos-exporter
 
-EXPOSE 8080
+RUN export GOOS=$(echo ${TARGETPLATFORM} | cut -d / -f1) && \
+    export GOARCH=$(echo ${TARGETPLATFORM} | cut -d / -f2)
 
-CMD ["sh", "-c","$EXPORTER"]
+RUN go env
 
+RUN CGO_ENABLED=0 go build -o /output/chaos-exporter -v ./cmd/exporter/
+
+# Packaging stage
+FROM alpine:latest
+
+LABEL maintainer="LitmusChaos"
+
+COPY --from=builder /output/chaos-exporter /
+
+RUN addgroup -S litmus && adduser -S -G litmus 1001
+
+USER 1001
+
+CMD ["./chaos-exporter"]
+
+EXPOSE 8080
```

Comment:
Should we expose it before `CMD`? Did you also test the nonroot changes?

Reply:
Non, necessary it will after that also

Reply:
I tested it. Working as non-root. We are following the same steps in litmus-portal